src/main/java/org/apache/sling/feature/impl/OrderedDictionary.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.feature.impl;
+
+import java.util.*;
+
+/**
+ * A dictionary implementation with predictable iteration order.
+ *
+ * Actually this class is a simple adapter from the Dictionary interface
+ * to a synchronized LinkedHashMap
+ *
+ * @param <K>
+ * @param <V>
+ */
+public class OrderedDictionary<K, V> extends Dictionary<K, V> implements Map<K, V> {
+    private static class EnumarationImpl<E> implements Enumeration<E> {
+        private final Iterator<E> iterator;
+
+        public EnumarationImpl(Iterator<E> iterator) {
+            this.iterator = iterator;
+        }
+
+        @Override
+        public boolean hasMoreElements() {
+            return iterator.hasNext();
+        }
+
+        @Override
+        public E nextElement() {
+            return iterator.next();
+        }
+    }
+
+    private Map map = Collections.synchronizedMap(new LinkedHashMap());
```

Comment:
I would be in favour of switching to `LinkedHashMap`, but the baseline plugin complained that I need to bump the major version and I thought I should try like this first.

Reply:
Did not see what we return it using a getter. Yes, that's a problem if changing to a LinkedHashMap, and probably @bosschaert or @cziegeler can tell us if that's a problem.

For the thread safety - I'm not sure we need it. A `Hashtable` is thread-safe but a `Dictionary` does not have to be. So IMO we could go with either a `LinkedHashMap`, while breaking backwards compatibility, or with a non-thread-safe `Dictionary` implementation backed by a `LinkedHashMap`.

Reply:
We're still at version 0.1.2 which means we didn't do a 1.0 release yet. In OSGi versions starting with 0.x.x indicate 'beta' releases and make no guarantees about backward compatibility, so I think we can apply that reasoning here too.

So I think we can still make things better without being 100% compatible. 
The previous use of Dictionary is probably somehow related to the fact that in OSGi configuration properties are stored in Dictionary objects (because this spec predates Java Collections) but I think that where we need Dictionaries we can quite easily convert to them there.

So from me +1 for switching to LinkedHashMap.